src/app/api/quality_matrix/replication_source.py

```diff
@@ -101,12 +101,12 @@ def queried_missing_properties(
     ).execute()
 
 
-def join_data(data: dict, key: str) -> dict[str, Union[str, dict]]:
-    return {"metadatum": key, "columns": data}
+def build_quality_output(data: dict, key: str) -> QualityOutput:
+    return QualityOutput(metadatum=key, columns=data, level=2)
 
 
-def api_ready_output(raw_input: dict) -> QUALITY_MATRIX_RETURN_TYPE:
-    return [join_data(data, key) for key, data in raw_input.items()]
+def api_ready_output(raw_input: dict) -> list[QualityOutput]:
```

Comment:
same here, just remove it and inline it to `_quality_matrix` where it is used.

Reply:
Inlined